src/app/embed/adapters/feed.adapter.ts

```diff
@@ -97,9 +97,12 @@ export class FeedAdapter implements DataAdapter {
     }
   }
 
-  parseFeedEpisodes(doc: XMLDocument, numEpisodes: number): AdapterProperties[] {
+  parseFeedEpisodes(doc: XMLDocument, numEpisodes: number | string): AdapterProperties[] {
     let episodes = Array.from(doc.querySelectorAll('item'));
-    return episodes.slice(0, numEpisodes).map((item, index) => {
+    if (typeof numEpisodes === 'number') {
```

Comment:
(Accidentally made this comment in the commit instead of the PR) My param comes in as the string "10" so it's always showing all. Maybe try !isNaN(+numEpisodes) instead.